math/iterative_factorial.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief Iterative implementation of [Factorial](https://en.wikipedia.org/wiki/Factorial)
+ *
+ * @author [Renjian-buchai](https://github.com/Renjian-buchai)
+ *
+ * @details Calculates factorial iteratively.
+ * \f[n! = n\times(n-1)\times(n-2)\times(n-3)\times\ldots\times3\times2\times1
+ *       = n\times(n-1)!\f]
+ * for example:
+ * \f$4! = 4\times3! = 4\times3\times2\times1 = 24\f$
+ *
+ * @example
+ *
+ * 5! = 5 * 4 * 3 * 2 * 1
+ *
+ * Recursive implementation of factorial pseudocode:
+ *
+ * function factorial(n):
+ *     if n == 1:
+ *         return 1
+ *     else:
+ *         return factorial(n-1)
+ *
+ */
+
+#include <cassert>    /// for assert
+#include <exception>  /// for std::invalid_argument
+#include <iostream>   /// for std::cout
+
+/**
+ * @namespace
+ * @brief Mathematical algorithms
+ */
+namespace math {
+
+/**
+ * @brief Calculates the factorial iteratively.
+ * @param n Nth factorial.
+ * @return Factorial.
+ * @note 0! = 1.
+ * @warning Maximum=20 because there are no 128-bit integers in C++. 21!
+ * returns 1.419e+19, which is not 21! but (21! % UINT64_MAX).
+ */
+uint64_t iterativeFactorial(uint8_t n) {
+    if (n > 20) {
+        throw new std::invalid_argument("Maximum n value is 20");
+    }
+
+    // 1 because it is the identity number of multiplication.
+    uint64_t accumulator = 1;
+
+    while (n > 0) {
+        accumulator *= n;
+        --n;
+    }
```

Comment:
Minor nit (non-blocking): It could just as easily be 
```suggestion
    while (n > 1) {
        accumulator *= n;
        --n;
    }
```
since multiplying by 1 has no effect 😄